gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/ThrowableFailureReporter.java

```diff
@@ -27,18 +28,23 @@ public final class ThrowableFailureReporter {
 
     public static <T extends Task> FailureReport getFailureReport(T task) {
         Throwable throwable = task.getState().getFailure();
-        // try to get the last ExtraInfoException in the causal chain
-        Optional<ExceptionWithSuggestion> maybeExtraInfoException = Throwables.getCausalChain(throwable).stream()
+        // try to get the last ExceptionWithSuggestion in the causal chain
+        Optional<ExceptionWithSuggestion> maybeExceptionWithSuggestion = Throwables.getCausalChain(throwable).stream()
                 .filter(ExceptionWithSuggestion.class::isInstance)
                 .map(ExceptionWithSuggestion.class::cast)
                 .findFirst();
-        return maybeExtraInfoException
+        Optional<MinimalException> maybeMinimalException = Throwables.getCausalChain(throwable).stream()
+                .filter(MinimalException.class::isInstance)
+                .map(MinimalException.class::cast)
+                .findFirst();
+        return maybeExceptionWithSuggestion
                 .map(exception -> getEnhancedExceptionReport(task.getPath(), throwable, exception))
-                .orElseGet(() -> getGenericExceptionReport(task, throwable));
+                .orElseGet(() -> maybeMinimalException
+                        .map(exception -> getMinimalExceptionReport(task.getPath(), exception))
+                        .orElseGet(() -> getGenericExceptionReport(task, throwable)));
```

Comment:
I'd probably do `.or(() -> maybeMinimalException.map(...).orElseGet(() -> generic` to avoid the slightly odd nesting.